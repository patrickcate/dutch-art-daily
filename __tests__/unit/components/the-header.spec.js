import { shallowMount } from '@vue/test-utils'
import TheHeader from '@components/the-header.vue'

describe('TheHeader Component', () => {
  it('is a Vue instance', () => {
    const wrapper = shallowMount(TheHeader)
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    const wrapper = shallowMount(TheHeader)
    expect(wrapper).toMatchSnapshot()
  })

  it('shows share list on larger screen sizes', () => {
    const wrapper = shallowMount(TheHeader, {
      computed: {
        breakpoint() {
          return {
            md: true,
          }
        },
      },
    })

    expect(wrapper.find({ name: 'ShareList' }).isVisible()).toBe(true)
    expect(wrapper.find({ name: 'ShareDropdown' }).isVisible()).toBe(false)
  })

  it('shows share dropdown on smaller screen sizes', () => {
    const wrapper = shallowMount(TheHeader, {
      computed: {
        breakpoint() {
          return {
            md: false,
          }
        },
      },
    })

    expect(wrapper.find({ name: 'ShareList' }).isVisible()).toBe(false)
    expect(wrapper.find({ name: 'ShareDropdown' }).isVisible()).toBe(true)
  })
})
