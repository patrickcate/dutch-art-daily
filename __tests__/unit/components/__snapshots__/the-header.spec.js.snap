// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHeader Component renders correctly 1`] = `
<header role="banner" class="header">
  <the-logo-stub class="header__branding"></the-logo-stub>
  <h1 class="header__date">
    <the-date-stub></the-date-stub>
  </h1>
  <share-list-stub direction="horizontal" class="header__share" style="display: none;"></share-list-stub>
  <share-dropdown-stub class="header__share" style="display: none;"></share-dropdown-stub>
</header>
`;
